csh-pings-backend/src/main/java/edu/rit/csh/pings/external/DiscordService.java

```diff
@@ -0,0 +1,121 @@
+package edu.rit.csh.pings.external;
+
+import edu.rit.csh.pings.entities.Application;
+import edu.rit.csh.pings.entities.DiscordServiceConfiguration;
+import edu.rit.csh.pings.entities.Route;
+import edu.rit.csh.pings.entities.VerificationRequest;
+import edu.rit.csh.pings.managers.VerificationRequestManager;
+import lombok.RequiredArgsConstructor;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.JDABuilder;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.requests.GatewayIntent;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.io.IOError;
+import java.io.IOException;
+import java.util.*;
+
+import static edu.rit.csh.pings.util.Util.readFully;
+
+@Service
+@RequiredArgsConstructor
+public class DiscordService implements ExternalService<DiscordServiceConfiguration> {
+
+    private final Log log = LogFactory.getLog("pings.discord_service");
+
+    private transient final Map<String, User> usernameToId = new HashMap<>();
+
+    private String pingTemplate;
+    private String verificationTemplate;
+    private final VerificationRequestManager verificationRequestManager;
+
+    @Value("${csh.pings.discord.bot_id}")
+    private String DISCORD_TOKEN;
+
+    @Value("${csh.pings.url:https://pings.csh.rit.edu/}")
+    private String url;
+
+
+    private final Collection<GatewayIntent> intents = EnumSet.of(GatewayIntent.GUILD_MEMBERS, GatewayIntent.GUILD_MESSAGES);
+    private JDA jda = JDABuilder.create(DISCORD_TOKEN, intents).build();
+
+    @PostConstruct
+    public void readTemplates() {
+        try {
+            this.pingTemplate = readFully(new ClassPathResource("discord_ping_template.txt").getInputStream());
+            this.verificationTemplate = readFully(new ClassPathResource("discord_verification_template.txt").getInputStream());
+        } catch (IOException e) {
+            throw new IOError(e);
+        }
+    }
+
+    @PostConstruct
+    public void setup() {
+        try {
+            jda.awaitReady();
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        // get the guilds the bot is in
+        jda.getGuilds().forEach(guild -> {
+            // get the users in the guild
+            guild.getMembers().forEach(member -> {
+                // add the user to the set
+                usernameToId.put(member.getUser().getName() + "#" + member.getUser().getDiscriminator(), member.getUser());
+            });
+        });
+        sendMessageTest("Creeper", "3621", "Discord bot is online!");
```

Comment:
Do you intend to send yourself (I'm assuming) a message every time the bot is created / destroyed?